client/src/pages/pot-instance/components/SimpleTimerView/SimpleTimerView.tsx

```diff
@@ -0,0 +1,47 @@
+/* eslint-disable @typescript-eslint/no-unused-vars */
+/* eslint-disable no-unused-vars */
+import { Dispatch, SetStateAction, useEffect, useState } from 'react';
+import StackedProgressBar from 'components/StackedProgressBar';
+import { PotContent, ProgressData } from 'pages/pot-instance/models';
+import getRGBPerCategory from 'utils/colors';
+
+interface Props {
+  potContent: PotContent[];
+  setPotContent: Dispatch<SetStateAction<PotContent[]>>;
+}
+
+interface Data extends PotContent {
+  color: string;
+}
+
+const SimpleTimerView = ({ potContent, setPotContent }: Props) => {
+  console.log(setPotContent);
+  const [data, setData] = useState<Map<string, Data>>(new Map());
+
+  const [outputData, setOutData] = useState<ProgressData[]>([]);
+  useEffect(() => {
+    potContent.forEach((potItem) => {
+      data.set(potItem.id, {
+        ...potItem,
+        color: data.get(potItem.id)?.color || getRGBPerCategory()
+      });
+    });
```

Comment:
Optimization question: Does this useEffect re-run each item in PotContent whenever it changes? Or does it just run for only new items added to potContent?

I can check/fiddle with it when I get home and run the component

Reply:
It will execute each time the timeLeft decreases as well.